src/Provider/Exception/ResponseParsingException.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * This file is part of the league/oauth2-client library
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright Copyright (c) Alex Bilbie <hello@alexbilbie.com>
+ * @license http://opensource.org/licenses/MIT MIT
+ * @link http://thephpleague.com/oauth2-client/ Documentation
+ * @link https://packagist.org/packages/league/oauth2-client Packagist
+ * @link https://github.com/thephpleague/oauth2-client GitHub
+ */
+
+namespace League\OAuth2\Client\Provider\Exception;
+
+use Exception;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Exception thrown if the parser cannot parse the provider response.
+ */
+class ResponseParsingException extends Exception
+{
+    /**
+     * @var ResponseInterface
+     */
+    protected $response;
+
+    /**
+     * @var string
+     */
+    protected $responseBody;
+
+    /**
+     * @param ResponseInterface $response The response
+     * @param string $responseBody The response body
+     * @param null $message
+     * @param int $code
+     */
+    public function __construct(
+        ResponseInterface $response,
+        $responseBody,
+        $message = null,
+        $code = 0
```

Comment:
Add another parameter here for `Throwable $previous = null`.

Reply:
`Throwable` is introduced in php version 7. Should we keep support for version 5.6, we should use Exception. However, I do see a function used in AbstractProvider which is not supported by 5.6 e.g., random_bytes.

Reply:
Good point on `Throwable`. Use `Exception` for now.

We are continuing to support PHP 5.6 for the 2.x versions. We're able to use `random_bytes()` in 5.6 through a polyfill (`paragonie/random_compat`).